Team/fried cucumber/feature/character collider

rastkomiocinovic opened the pull request.

rastkomiocinovic committed:
Added collision between the player and walls.
Generating collision test world.

rastkomiocinovic committed:
Tweeked player collider to better fit into the world.
Added different tiles to wall ends.

DPrivalov commented on source/dagger/gameplay/team_game/team_game_main.cpp:
```diff
@@ -65,6 +73,11 @@ struct MainCharacter
+        //Collision setup:        
+        chr.col.size = { 25, 28 };
+        chr.tm.position = { position_ , 0.0f };
+        CharacterCollisionSystem::MainCharacterEntity = entity;
```
Just a suggestion:
instead of doing this you can create a MainCharacter component (empty struct) and then emplace it here. And in any place you need you can access it by 
auto view = Engine::Registry().view<MainCharacter>();

rastkomiocinovic committed:
Generalized collisions - added generic collisions.
Character-Wall collisions are now a special case.
Filenames changed to reflect changes.

Nikolar97 committed:
Merge branch 'team/fried_cucumber/main' into team/fried_cucumber/feature/character_collider
